src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs

```diff
@@ -65,22 +66,27 @@ public FileStream(IntPtr handle, FileAccess access, bool ownsHandle, int bufferS
         private static void ValidateHandle(SafeFileHandle handle, FileAccess access, int bufferSize, bool isAsync)
         {
             if (handle.IsInvalid)
+            {
                 throw new ArgumentException(SR.Arg_InvalidHandle, nameof(handle));
```

Comment:
Is there any value in replacing all the remaining `throw`s with `ThrowHelper`s in `FileStream` code? Just curious, not asking for the change.

Reply:
The main purpose of using `ThrowHelper` is to move the `throw` to a separate method and get the caller inlined for methods that are on the hot path. 

I've initially used the `ThrowHelper` in methods that were small and not inlined and after that ensured that the code is consistent. By consistent, I mean that if we throw one specific exception in `System.IO`, it should be thrown in the same way everywhere (by for example using the same `ThrowHelper` method).

> Is there any value in replacing all the remaining throws with ThrowHelpers in FileStream code?

I think that if we profile the code and see that the given method is not getting inlined, while it should then yes. Otherwise no.